src/amp-story-player/amp-story-player-impl.js

```diff
@@ -870,13 +891,52 @@ export class AmpStoryPlayer {
    * @param {!Element} story
    * @param {!Element} iframe
    * @param {!VisibilityState} visibilityState
+   * @return {!Promise}
    * @private
    */
   layoutIframe_(story, iframe, visibilityState) {
-    this.maybeGetCacheUrl_(story.href).then((url) => {
-      const {href} = this.getEncodedLocation_(url, visibilityState);
-      iframe.setAttribute('src', href);
-      iframe.setAttribute('title', story.textContent.trim());
+    return this.hasSameHref_(story.href, iframe.src).then((hasBeenLaidOut) => {
```

Comment:
This could get much easier if you'd keep the previous code to retrieve the cacheURL and comparing it to the iframe src right here.

ie:

```
this.maybeGetCacheUrl_(story.href).then((url) => {
  if (sanitizedUrlsAreEquals) return;
});
```

Then you wouldn't have to retrieve the cache URL a second time.